### PR TITLE
The Message Broker

### LINKED ISSUE #310: Message Broker Payload
Payload to send to Message Broker for **Domestic** (US):
```
$payload = array(
  'email' => String, (optional),
  'mobile' => Int (required),
  'mc_opt_in_path_id' => 180170,
  'mobile_tags' => array(
    'agg',
    'xxx' (candidate_id),
    'candidate_name'
    'candidate_gender'
  ),
  'drupal_uid' => Int (assuming app does user registration via NorthStar),
  'first_name' => String,
  'birthdate_timestamp' => Timestamp,
  'country_code' => 'US',
  'candidate_name' => String (vote),
  'cause_theme' => String,
  'activity' => 'vote',
  'activity_timestamp' => Timestamp,
  'application_id' => 'AGG'
);
```
**International**:
```
$payload = array(
  'email' => String (required),
  'email_template' => "agg2015-voting-confirmation-global-non-affiliates",
  'email_tags' => array(
    'agg',
    'xxx' (candidate_id),
    'candidate_name'
    'candidate_gender'
  ),
  'mailchimp_list_id" => "f2fab1dfd4",
  'mailchimp_grouping_id" => 10677,
  'mailchimp_group_name" => "AGG2015",
  'merge_vars' => array(
    'FNAME' =>  String (first name)
  ),
  'mobile' => Int (optional),
  'mobile_tags' => array(
    'agg',
    'candidate_id',
  ), (optional)
  'drupal_uid' => Int (assuming app does user registration via NorthStar),
  'first_name' => String,
  'birthdate' => Timestamp,
  'country_code' => String (2 characters),
  'candidate_name' => String (vote),
  'cause_theme' => String,
  'subscribed' => 1,
  'activity' => 'vote',
  'activity_timestamp' => Timestamp,
  'application_id' => 'AGG'
);
```

**Related**:
- https://github.com/DoSomething/mb-logging-api/issues/29

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes
- Updated message broker payload via Dee's instruction. Closes #310.
- Removes "welcome message" event, since we're only sending a single message on first vote.

For review: @angaither @DeeZone

### CODE REVIEW

File: app/Handlers/Events/SendFirstVoteMessage.php [hunk 1]
```diff
@@ -39,21 +35,38 @@ public function handle(UserCastFirstVote $event)
             'candidate_name' => $event->candidate->name,
```
Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: Please add `candidate_id => $event->candidate->id` in addition to the tag setting.

File: app/Handlers/Events/SendFirstVoteMessage.php [hunk 2]
```diff
@@ -39,21 +35,38 @@ public function handle(UserCastFirstVote $event)
...
-            'mc_opt_in_path_id' => env('MC_OPT_IN_PATH'),
-            'merge_vars' => [
-                'FNAME' => $event->user->first_name
-            ]
         ];
 
+        // Send fields for domestic users
+        if($event->user->country_code === 'US') {
+            $payload['mobile'] = $event->user->phone;
+            $payload['mc_opt_in_path_id'] = env('MC_OPT_IN_PATH');
```
Comment (DeeZone) [inline app/Handlers/Events/SendFirstVoteMessage.php]: Remove this

File: app/Handlers/Events/SendFirstVoteMessage.php [hunk 3]
```diff
@@ -39,21 +35,38 @@ public function handle(UserCastFirstVote $event)
...
+            $payload['mailchimp_grouping_id'] = env('MAILCHIMP_GROUP_ID');
+            $payload['mailchimp_group_name'] = env('MAILCHIMP_GROUP_NAME');
+            $payload['mailchimp_list_id'] = env('MAILCHIMP_LIST_ID');
+            $payload['email_template'] = env('VOTE_TEMPLATE', 'mb-votingapp-vote');
+            $payload['email_tags'] = [
+                env('APP_NAME_TAG', 'votingapp'),
+                $event->candidate->id
+            ];
+            $payload['merge_vars'] = [
+                'FNAME' => $event->user->first_name
```
Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php]: Add:

```
$payload['merge_vars'] = [
    'FNAME' => $event->user->first_name,
    'CANDIDATE_NAME' => $event->candidate->name,
    'CANDIDATE_ID' => $event->candidate->id
  ];
```

**Comment by angaither** [comment]
:+1: looks ok to me...but will default to @DeeZone here

File: app/Handlers/Events/SendFirstVoteMessage.php (hunk 1 above)
Comment (DFurnes) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: I'm a bit confused. Where should I be adding that line?

Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: Plus this one ^^

Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: Sorry about that, try this:

```
...
'candidate_name' => $event->candidate->name,
'candidate_id' => $event->candidate->id,
...
```

Comment (DFurnes) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: `'candidate_id' => $event->candidate->id` is on the line above. Is there somewhere else I should be adding it in the payload?

File: app/Handlers/Events/SendFirstVoteMessage.php [hunk 4]
```diff
@@ -39,21 +35,39 @@ public function handle(UserCastFirstVote $event)
...
+            $payload['mailchimp_group_name'] = env('MAILCHIMP_GROUP_NAME');
+            $payload['mailchimp_list_id'] = env('MAILCHIMP_LIST_ID');
+            $payload['email_template'] = env('VOTE_TEMPLATE', 'mb-votingapp-vote');
+            $payload['email_tags'] = [
+                env('APP_NAME_TAG', 'votingapp'),
+                $event->candidate->id
+            ];
+            $payload['merge_vars'] = [
+                'FNAME' => $event->user->first_name,
+                'CANDIDATE_NAME' => $event->candidate->name
```
Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php]: Is there any value  in a link back to the candidate page to perhaps check on the number of votes they have?

Comment (DFurnes) [inline app/Handlers/Events/SendFirstVoteMessage.php]: Users are not able to see number of votes, but they can return to vote again every 24 hours, or share the page via Facebook or Twitter or whatnot. So perhaps? Happy to send it along if you think you'll use it. :)

File: app/Handlers/Events/SendFirstVoteMessage.php (hunk 1 above)
Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: Ah, my bad, needed a page refresh. Good as is.

Comment (DFurnes) [inline app/Handlers/Events/SendFirstVoteMessage.php:36]: Cursed :octocat:!

File: app/Handlers/Events/SendFirstVoteMessage.php (hunk 4 above)
Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php]: If it's not a big lift:

```
  'CANDIDATE_LINK' => $candidate->link
```

Comment (DFurnes) [inline app/Handlers/Events/SendFirstVoteMessage.php]: Added!

screen shot 2015-06-02 at 3 14 18 pm

**Comment by deezone** [comment]
:+1:

**Comment by DFurnes** [comment]
:clap: Merging! (And happy to make more changes if necessary in the future!)

**Pull request merged**